src/Microsoft.AspNetCore.Server.IISIntegration.Tools/PublishIISCommand.cs

```diff
@@ -67,9 +73,14 @@ private string GetApplicationBasePath()
             return Directory.GetCurrentDirectory();
         }
 
-        private string GetApplicationName(string applicationBasePath)
+        private static ProjectContext GetProjectContext(string applicationBasePath, string framework)
         {
-            return ProjectReader.GetProject(Path.Combine(applicationBasePath, "project.json")).Name;
+            var project = ProjectReader.GetProject(Path.Combine(applicationBasePath, "project.json"));
+
+            return new ProjectContextBuilder()
+                .WithProject(project)
+                .WithTargetFramework(framework)
+                .Build();
         }
 
         private static bool ConfigureForAzure()
```

Comment:
Is there a reason ConfigureForAzure can't be specified via an optional command line arg? Having the env variable too is OK, but it seems strange that you can't just specify this directly.